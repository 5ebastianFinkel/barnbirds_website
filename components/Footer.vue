<!--

This is the footer component including the copyright and a link to the 'Impressum'

Author: sfinkel
Created: 2021/10/15 | 22:08 CET

-->
<template>
  <div class="container">
    <span>
    <nuxt-link data-testid="impressum" to="/Impressum">Impressum</nuxt-link><span class="text">| Copyright &copy; 2021-heute The Barnbirds </span>
    </span>
  </div>
</template>

<script>
import { faCopyright } from '@fortawesome/free-regular-svg-icons'

export default {
  name: 'Footer',
  computed: {
    faCopyright () {
      return faCopyright
    }
  }
}
</script>

<style scoped>

.container {
  position: relative;
  bottom: 0;
  right: 0;
  left: 0;
  z-index: var(--z-index-navbar);
  display: flex;
  justify-content: center;
  align-items: center;
  border-top: 1px solid var(--c-divider);
  padding: 0.7rem 1.5rem 0.7rem 4rem;
  height: var(--header-height);
  background-color: var(--c-bg);
}

.text {
  margin: 0;
  text-align: center;
  line-height: 1.4;
  font-size: 0.9rem;
  color: var(--c-text-light);
}
</style>
