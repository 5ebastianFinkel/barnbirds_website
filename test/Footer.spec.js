import { mount, RouterLinkStub } from '@vue/test-utils'
import Footer from '@/components/Footer'

describe('Footer', () => {
  test('is a Vue instance', () => {
    const wrapper = mount(Footer, {
      stubs: {
        NuxtLink: RouterLinkStub
      }
    })
    const link = wrapper.findComponent(RouterLinkStub)
    expect(wrapper.vm).toBeTruthy()
    expect(link.element.textContent).toBe('Impressum')
  })

  test('renders correctly', () => {
    const wrapper = mount(Footer, {
      stubs: {
        NuxtLink: RouterLinkStub
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
