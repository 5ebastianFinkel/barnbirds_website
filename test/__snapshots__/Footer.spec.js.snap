// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<div
  class="container"
>
  <span>
    <a
      data-testid="impressum"
    >
      Impressum
    </a>
    <p
      class="text"
    >
      | Copyright Â© 2021-heute The Barnbirds 
    </p>
  </span>
</div>
`;
